src/external_calls.rs

```diff
@@ -178,7 +178,9 @@ pub fn run<
         out_of_circuit_vm.memory.execute_partial_query(0, query);
     }
 
-    let mut tracer = GenericNoopTracer::<_>::new();
+    // we are using TestingTracer to track prints and exceptions inside out_of_circuit_vm cycles
+    let mut tracer = TestingTracer::default();
```

Comment:
darn, I forgot one thing.

This method that we're in (run) is actually also used in production (long story).

So we must not change the behavior there.

Reply:
(that's also why I wanted 'testingtracer' to be under 'cfg(test)' - as then compiler tells you things like that ;-)

Reply:
one option is to do:

#[cfg(test)]
let mut tracer = TestingTracer::default();
#[cfg(not(test))]
let mut tracer = GenericTracerNoop;

Reply:
WDYT?

`#[cfg(test)]
fn get_tracer() {
    GenericNoopTracer::<_>::new()
}

#[cfg(not(test))]
fn get_tracer() {
    TestingTracer::default()
}`